### PR TITLE
Referral code credit history

### PR DESCRIPTION
**Pull request by scotthasbrouck**
Still needs tests

### CODE REVIEW

File: src/3_branch.js [hunk 1]
```diff
@@ -947,14 +933,13 @@ Branch.prototype['credits'] = wrap(callback_params.CALLBACK_ERR_DATA, function(d
...
 
-
-if (CORDOVA_BUILD) { // jshint undef:false
 /**
  * @function Branch.creditHistory
  * @param {Object} data - _optional_ - options controlling the returned history.
  * @param {function(?Error,Object=)=} callback - _required_ - returns an array with credit history data.
  *
  * This call will retrieve the entire history of credits and redemptions from the individual user.
+ * // wtf is "direction"?
```
Comment (dmitrig01) [inline src/3_branch.js]: 'asc' or 'desc' i think

let's rename 'data' to 'options' and actually document those - i think those are documented somewhere, probably in the iOS SDK or something?

Comment (scotthasbrouck) [inline src/3_branch.js]: Yeah I found it in the api docs. It is direction, so I clarified it in this documentation.

**Comment by scotthasbrouck** [comment]
@dmitrig01 forgot to tag you in this yesterday. Lmk how this looks now. I'm working on the banner/body positioning stuff in the other open PR and could release that with this together.

File: src/1_resources.js [hunk 2]
```diff
@@ -62,348 +62,263 @@ function validator(required, type) {
...
+			"sdk": validator(true, validationTypes.str),
+			"data": validator(false, validationTypes.str),
+			"alias": validator(false, validationTypes.str),
+			"tags": validator(false, validationTypes.arr),
+			"feature": validator(false, validationTypes.str),
+			"channel": validator(false, validationTypes.str),
+			"stage": validator(false, validationTypes.str),
+			"type": validator(false, validationTypes.num)
+		}
+	};
```
Comment (dmitrig01) [inline src/1_resources.js]: you oevrride this below anyway :-)

File: test/4_integration.js [hunk 3]
```diff
@@ -45,7 +45,7 @@ describe('Integration tests', function() {
 		branch.init(browser_fingerprint_id, callback);
 		if (assert) {
 			assert.equal(requests.length, 1);
-			assert.equal(requests[0].src, 'https://bnc.lt/_r?v=' + config.version + '&callback=branch_callback__' + jsonpCallback.toString());
+			assert.equal(requests[0].src, 'https://bnc.lt/_r?sdk=web' + config.version + '&callback=branch_callback__' + jsonpCallback.toString());
```
Comment (dmitrig01) [inline test/4_integration.js]: does this do sdk=web or sdk=web0.0.0?

Comment (dmitrig01) [inline test/4_integration.js]: oh never mind, i see it appends the version

**Comment by dmitrig01** [comment]
sweet, minus the one really minor thing, this looks good to go

**Pull request merged**